src/monmove.cpp

```diff
@@ -1570,3 +1576,69 @@ int monster::turns_to_reach( int x, int y )
 
     return static_cast<int>( turns + .9 ); // Halve (to get turns) and round up
 }
+
+void monster::shove_vehicle( const tripoint &remote_destination,
+                             const tripoint &nearby_destination )
+{
+    if( this->has_flag( MF_PUSH_VEH ) ) {
+        auto vp = g->m.veh_at( nearby_destination );
+        if( vp ) {
+            vehicle &veh = vp->vehicle();
+            units::mass veh_mass = veh.total_mass();
+            int shove_moves = 0;
+            int shove_velocity = 0;
+            float shove_damage_min = 0.00F;
+            float shove_damage_max = 0.00F;
+            switch( this->get_size() ) {
+                case MS_TINY:
+                case MS_SMALL:
+                    break;
+                case MS_MEDIUM:
+                    if( veh_mass < 500_kilogram ) {
+                        shove_moves = -50;
+                        shove_velocity = 500;
+                        shove_damage_min = 0.00F;
+                        shove_damage_max = 0.01F;
+                    }
+                    break;
+                case MS_LARGE:
+                    if( veh_mass < 1000_kilogram ) {
+                        shove_moves = -100;
+                        shove_velocity = 1000;
+                        shove_damage_min = 0.00F;
+                        shove_damage_max = 0.03F;
+                    }
+                    break;
+                case MS_HUGE:
+                    if( veh_mass < 1500_kilogram ) {
+                        shove_moves = -150;
```

Comment:
That means that hulks would shove the lightest shopping cart and the heavy car for the same 150 moves?
Also, this means that medium-sized brute will shove the cart for only 50 moves, while huge-sized hulk will shove the same cart for 150 moves?

Reply:
Move cost in independent from shoved vehicle size right now, but I guess we can change that to some weight-dependent function.

Reply:
Yeah, that would make sense.

Reply:
Done.